### ISSUE TITLE
Add avoid_unused_function_parameters

### ISSUE THREAD

**Issue by Brunobnahorny** [dart-lang/lints#87]
**Describe the rule you'd like to see added and to what rule set**
Like `avoid_unused_constructor_parameters`, unused parameters in function is surely unintentional

**Additional context**

```dart
badFunction(int unusedParameter, [String unusedPositional]) {
    print(unusedParameter);
}

**Comment by devoncarew** [comment]
If this a request for a new lint, or a request to add an existing lint to the package:lints set (new lint requests should instead go to dart-lang/linter)?

**Comment by Brunobnahorny** [comment]
Oh! my bad.
It is a new lint rule, thank you!
